shell/platform/windows/system_utils.cc

```diff
@@ -8,79 +8,48 @@
 
 #include <sstream>
 
+#include "flutter/fml/logging.h"
 #include "flutter/fml/platform/win/wstring_conversion.h"
 
 namespace flutter {
 
-std::vector<LanguageInfo> GetPreferredLanguageInfo(
-    const WindowsRegistry& registry) {
-  std::vector<std::wstring> languages = GetPreferredLanguages(registry);
+std::vector<LanguageInfo> GetPreferredLanguageInfo() {
+  std::vector<std::wstring> languages = GetPreferredLanguages();
   std::vector<LanguageInfo> language_info;
   language_info.reserve(languages.size());
 
   for (auto language : languages) {
+    FML_LOG(ERROR) << "Pushing " << fml::WideStringToUtf8(language);
     language_info.push_back(ParseLanguageName(language));
   }
+  FML_LOG(ERROR) << "Got " << language_info.size() << "langs";
   return language_info;
 }
 
-std::wstring GetPreferredLanguagesFromRegistry(const WindowsRegistry& registry,
-                                               ULONG buffer_size) {
-  std::wstring buffer(buffer_size, '\0');
-  if (registry.GetRegistryValue(HKEY_CURRENT_USER, kGetPreferredLanguageRegKey,
-                                kGetPreferredLanguageRegValue,
-                                RRF_RT_REG_MULTI_SZ, NULL, buffer.data(),
-                                &buffer_size) != ERROR_SUCCESS) {
-    return std::wstring();
-  }
-  return buffer;
-}
-
 std::wstring GetPreferredLanguagesFromMUI() {
   ULONG buffer_size;
   ULONG count = 0;
   DWORD flags = MUI_LANGUAGE_NAME | MUI_UI_FALLBACK;
   if (!GetThreadPreferredUILanguages(flags, &count, nullptr, &buffer_size)) {
```

Comment:
It looks like they are using the same method as here, there. And re: your comment on `WindowsProcTable`, this shouldn't be necessary. Looks like I missed a nuance of the parameters to the function that didn't make a difference locally but did on devicelab

Reply:
> It looks like they are using the same method as here, there

Ah you're right, I got confused by their helper's name.

> And re: your comment on WindowsProcTable, this shouldn't be necessary.

The current tests depend on the machines environment. For example, the tests would fail if somehow a machine was configured such that `GetThreadPreferredUILanguages` returns no languages.

In my mind, the ideal test would be an integration test that mocks `GetThreadPreferredUILanguages` and then checks that the Windows embedder sends the expected `FlutterLocale` to the embedder API at startup. Through our mocks we can guarantee that the machine's language configuration can't cause unexpected test flakes.

Reply:
> Are we sure that:
> This is a concern, e.g. that the environment can return no preferred languages?

This is a hypothetical scenario, yes. Another hypothetical scenario would be that `GetThreadPreferredUILanguages` fails and returns `false`. These hypothetical scenarios would cause flakes. To reduce this likelihood, our tests should be deterministic and we should control their inputs.

> Doing so will not obscure the purpose of this test?

My only concern is using the real `GetThreadPreferredUILanguages` implementation in the test. I'm flexible on the test itself, I'd be happy with your current test too if `GetThreadPreferredUILanguages` was mocked out :)